### PR TITLE
IDSEQ-1133 - Improve permissions on results folders

### PR DESCRIPTION
**Pull request by jshoe**
### Description
- Only the original uploader should be able to get to raw_results_folder (otherwise they would never get a link to it from results_folder either).
- See https://jira.czi.team/browse/IDSEQ-1133

### Tests
- See tests

### CODE REVIEW

File: spec/controllers/pipeline_viz_controller_spec.rb [hunk 1]
```diff
@@ -154,5 +154,21 @@
         expect(response).to have_http_status 401
       end
     end
+
+    describe "GET raw_results_folder (nonadmin)" do
```
Comment (jshoe) [inline spec/controllers/pipeline_viz_controller_spec.rb]: going to move to a different file..

File: app/controllers/samples_controller.rb [hunk 2]
```diff
@@ -947,9 +947,15 @@ def unidentified_fasta
...
-    @file_list = @sample.results_folder_files
-    @file_path = "#{@sample.sample_path}/results/"
-    render template: "samples/raw_folder"
+    if current_user.id == @sample.user_id
+      @file_list = @sample.results_folder_files
+      @file_path = "#{@sample.sample_path}/results/"
+      render template: "samples/raw_folder"
+    else
+      render json: {
+        message: "Only the original uploader can view the sample's raw results folder."
```
Comment (gregdingle) [inline app/controllers/samples_controller.rb]: how will this be communicated to the user? or should the never have a link to the raw results if they are not allowed to see them?

Comment (gregdingle) [reply app/controllers/samples_controller.rb]: nevermind I see

File: app/controllers/samples_controller.rb [hunk 3]
```diff
@@ -1265,4 +1268,13 @@ def sort_by(samples, dir = nil)
     end
     samples
   end
+
+  def check_owner
```
Comment (jshoe) [inline app/controllers/samples_controller.rb]: this is here instead of ApplicationController b/c it's samples-specific and to access @sample

**Comment by jshoe** [comment]
Updated to add OWNER_ACTIONS and add back admin access

**Pull request merged**